src/__tests__/__views__/Login.test.js

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+import { shallow } from "enzyme";
+import sinon from "sinon";
+import { Login } from "../../views/Login";
+import Validator from "../../utils/validator";
+import TextInput from "../../components/common/Inputs/TextInput";
+import FormButton from "../../components/common/Buttons/FormButton";
+import BasicButton from "../../components/common/Buttons/BasicButton";
+import SocialButton from "../../components/common/Buttons/SocialButton";
+
+const [handleTextInput, handleSignIn] = new Array(2).fill(jest.fn());
+jest.mock("../../utils/validator");
+
+const setUp = () => {
+  const props = {
+    isSubmitting: false,
+    email: "",
+    password: "",
+    errors: {},
+    successMessage: null,
+    token: null,
+    handleSignIn,
+    handleTextInput
+  };
+  const wrapper = shallow(<Login {...props} />);
+  return { props, wrapper };
+};
+describe("Login component", () => {
+  let component;
+  beforeEach(() => {
+    sinon.spy(Login.prototype, "handleOnChange");
+    sinon.spy(Login.prototype, "handleSubmit");
+    sinon.spy(Login.prototype, "handleNavigation");
+    const { wrapper } = setUp();
+    component = wrapper;
+  });
+  afterEach(() => {
+    Login.prototype.handleOnChange.restore();
+    Login.prototype.handleSubmit.restore();
+    Login.prototype.handleNavigation.restore();
+  });
+  describe("component render", () => {
+    it("should render without an error", () => {
+      expect(component.find(`[data-test="login"]`).length).toBe(1);
+    });
+    it("should render two part of the screen", () => {
+      expect(component.find(`[data-test="auth-left"]`).length).toBe(1);
+      expect(component.find(`[data-test="auth-right"]`).length).toBe(1);
+    });
+    it("should render all basic components", () => {
+      expect(component.find(`[data-test="logo"]`).length).toBe(2);
+      expect(component.find(`[data-test="login-form"]`).length).toBe(1);
+      expect(component.find(`[data-test="nav-link"]`).length).toBe(2);
+      expect(component.find(SocialButton).length).toBe(3);
+    });
+    it("should render two TextInputs and button", () => {
+      expect(component.find(TextInput).length).toBe(2);
+      expect(component.find(FormButton).length).toBe(1);
+    });
+    it("should navigate to sign up page", () => {
+      component
+        .find(BasicButton)
+        .at(0)
+        .simulate("click");
+      expect(Login.prototype.handleNavigation.calledOnce).toBe(true);
+    });
+    it("should render component with initial props", () => {
+      component.find(TextInput).forEach(input => {
+        expect(input.props().value).toBe("");
+      });
+    });
+    it("should call handleOnChange when user enter input", () => {
```

Comment:
Codeclimate is complaining that you are repeating yourself!
 Can you create a function that simulates the OnChange event on this field and pass the target field and the value as a parameter?